### PR TITLE
Add undo sending for new messages

### PR DESCRIPTION
**Pull request by GretaD**
To do:

- [x] Show the undo dialog that undo sending of a message
- [x] Hide the composer modal while the undo dialog pop us is shown
- [x] Clicking the undo should show the last state of the message before sending
- [x] Move modal composer into a vuex store to use it globally #6181 
- [x] Start a timer when the message is fake-sent and stop it if the user clicks *Undo*
- [x] Requires https://github.com/nextcloud/mail/pull/6216

Fixes https://github.com/nextcloud/mail/issues/5438

### CODE REVIEW

File: src/components/NewMessageModal.vue [hunk 1]
```diff
@@ -33,6 +35,7 @@ export default {
 		return {
 			original: undefined,
 			originalBody: undefined,
+			hideComposer: false,
```
Comment (ChristophWurst) [inline src/components/NewMessageModal.vue:38]: unused

**Review by ChristophWurst — APPROVED** [review]
Works for me both for new messages and the ones that are already in the outbox

File: src/components/NewMessageModal.vue [hunk 2]
```diff
@@ -23,19 +24,17 @@ import logger from '../logger'
 import { toPlain } from '../util/text'
 import { saveDraft } from '../service/MessageService'
 import Composer from './Composer'
+import { showError, showUndo } from '@nextcloud/dialogs'
 import { translate as t } from '@nextcloud/l10n'
 
+const UNDO_DELAY = 7 * 1000
```
Comment (skjnldsv) [inline src/components/NewMessageModal.vue:30]: TODO: could be exposed from the capabilities
Maybe a followup for cleaner code :)

Comment (ChristophWurst) [reply src/components/NewMessageModal.vue:30]: what capabilities?

Comment (skjnldsv) [reply src/components/NewMessageModal.vue:30]: That UNDO is supported and the timeout which the backend will use before sending the message?
Or is it being sen straight from the front only?

Comment (ChristophWurst) [reply src/components/NewMessageModal.vue:30]: Straight outta frontend. The backend only sends the message if the tab/browser is closed

Comment (ChristophWurst) [reply src/components/NewMessageModal.vue:30]: Nevertheless I forgot to add the delay to the message's `sendAt`, so the backend could have picked up the message before the elapsed undo timer :see_no_evil: 

Fixed.

**Comment by JuliaKirschenheuter** [comment]
Test case:
1. Create  a new message
2. Send this message
3. Undo
4. Close a composer-dialog
5. Click to ... in a message-menu -> "Send now"
6. Expect message to be sent, but it will not be sent, outbox counter is still old one

**Comment by ChristophWurst** [comment]
Right. That is broken.

**Comment by ChristophWurst** [comment]
- [ ] BUG: somehow the message has an `aliasId` equal to the `accountId` and the lookup fails on the server, hence the message can't be sent. We have to debug how that happens.

**Comment by ChristophWurst** [comment]
> somehow the message has an `aliasId` equal to the `accountId`

I wonder how that happened @miaulalala? https://github.com/nextcloud/mail/blob/8627531f617f2674f2ccd27c678aa62ba8d8a9c5/lib/Db/LocalMessage.php#L103 :see_no_evil:

**Review by JuliaKirschenheuter — APPROVED** [review]

**Pull request merged**